src/main/java/org/rumbledb/expressions/flowr/ForClauseVar.java

```diff
@@ -53,10 +53,9 @@ public ForClauseVar(
         // because TreatIterator is wrapping the whole assignment expression,
         // meaning there is not one TreatIterator for each variable we loop over.
         if (sequenceType != null)
```

Comment:
I believe this sequenceType != null check is redundant now. Super()'s constructor ensures that sequence type should not be null

Reply:
Done.